src/test_workflow/perf_test/perf_test_cluster.py

```diff
@@ -57,22 +46,31 @@ def start(self):
         subprocess.check_call(command, cwd=os.getcwd(), shell=True)
         with open(self.output_file, "r") as read_file:
             load_output = json.load(read_file)
-        self.private_ip = load_output[self.stack_name]["PrivateIp"]
-        logging.info(f"Private IP: {self.private_ip}")
-        self.public_ip = load_output[self.stack_name].get("PublicIp", None)
+            self.create_endpoint(load_output)
+
+    def create_endpoint(self, cdk_output):
+        scheme = "https://" if self.cluster_config.security else "http://"
+        if self.cluster_config.is_single_node_cluster():
+            private_ip = cdk_output[self.stack_name]["PrivateIp"]
+            public_ip = cdk_output[self.stack_name].get("PublicIp", None)
+            self.is_endpoint_public = public_ip is not None
+            host = public_ip if public_ip is not None else private_ip
+        else:
+            host = cdk_output[self.stack_name]["LoadBalancerEndpoint"]
+            self.is_endpoint_public = True
+
+        if host is not None:
+            self.endpoint = host
+            self.cluster_endpoint_with_port = "".join([scheme, host, ":", str(self.port())])
+
+    def endpoint_with_port(self):
```

Comment:
Make this a `@property` so you can use it without `()`.

Reply:
done.